logger/common.go

```diff
@@ -117,34 +116,35 @@ func NewInfo(containerID string, containerName string, options ...InfoOpt) *dock
 
 // Start starts the actual logger.
 func (l *Logger) Start(uid int, gid int, cleanupTime *time.Duration, ready func() error) error {
-	var wg sync.WaitGroup
+	errChan := make(chan error)
 	if l.Stdout != nil {
-		wg.Add(1)
-		go l.sendLogs(l.Stdout, &wg, sourceSTDOUT, uid, gid, cleanupTime)
+		go l.sendLogs(errChan, l.Stdout, sourceSTDOUT, uid, gid, cleanupTime)
 	}
 	if l.Stderr != nil {
-		wg.Add(1)
-		go l.sendLogs(l.Stderr, &wg, sourceSTDERR, uid, gid, cleanupTime)
+		go l.sendLogs(errChan, l.Stderr, sourceSTDERR, uid, gid, cleanupTime)
 	}
```

Comment:
What about the case where `l.Stdout` and `l.Stderr` are both `nil`? In this case, `l.sendLogs()` is never invoked and L132 will also never unblock. Can this case ever happen?

Reply:
yeah, technically that should not happen, but it's still worth adding a check here. I will update in the next rev.

Reply:
I agree that it's worth logging an error message if both `l.Stdout` and `l.Stderr` are `nil`. More logging will definitely help us debugging shim logger related issues.